### PR TITLE
Port to Android

### PR DESCRIPTION
**Pull request by finagolfin**
Also gets the tests running on linux again.

I used an earlier version of this pull when packaging CookCLI for Android last weekend, termux/termux-packages#11081. I tested this pull by running `swift test` on Ubuntu 20.04 x86_64 and Android 12 AArch64 without a problem.

### CODE REVIEW

File: Tests/EmbassyTests/TestingHelpers.swift [hunk 1]
```diff
@@ -96,7 +102,7 @@ func makeRandomString(_ length: Int) -> String {
     let endIndex = UInt32(letters.count - 1)
     let result: [Any?] = Array(repeating: nil, count: length)
     return String(result.map({ _ in
-        letters[String.Index(encodedOffset: Int(arc4random_uniform(endIndex)))]
+        letters[String.Index(encodedOffset: Int(randomUniform(endIndex)))]
```
Comment (buttaface) [inline Tests/EmbassyTests/TestingHelpers.swift:105]: This was changed in b21f401 and broke on linux.

**Comment by buttaface** [comment]
Pinging @Goos, what do you think?

**Comment by buttaface** [comment]
@gregjlee, still involved?

**Comment by buttaface** [comment]
@beckychristensen, I notice you got another pull in recently, mind looking at this one?

**Comment by buttaface** [comment]
Ping @Goos, still looking to get this in.

**Comment by finagolfin** [comment]
Ping @Goos, I see a few other pulls have been merged since I opened this, any reason I've gotten no response on this?

**Comment by finagolfin** [comment]
Closing since a year has gone by with no response from the code owners.

**Pull request closed without merging**